pkg/render/intrusiondetection/dpi/dpi.go

```diff
@@ -95,10 +93,6 @@ func (d *dpiComponent) Objects() (objsToCreate, objsToDelete []client.Object) {
 	}
 	if d.cfg.HasNoDPIResource || d.cfg.HasNoLicense {
 		toDelete = append(toDelete, d.dpiAllowTigeraPolicy())
-		toDelete = append(toDelete, &corev1.Secret{
```

Comment:
Why don't we want to delete this any more?

Reply:
I removed all creations of it so if it's no longer being created it seems redundant to have deletion logic for something that isn't created. I'm realizing now that those creations were never in the `DeepPacketInspectionNamespace` so I'm not actually sure if or even how this is ever created. I can try to track it down.

Reply:
Stopping the creation of something can be done independently from deleting it - clusters that have had it created by a past version may still need for it to be deleted on upgrade, even if the newer version no longer creates it.

If anything, we should start deleting this unconditionally so that it's always removed if it's no longer needed at all (and then at some future point we can remove the deletion once we're confident that we don't need to support direct upgrade from a version that had previously created this)